exercises/space-age/src/example.c

```diff
@@ -1,6 +1,43 @@
 #include "space_age.h"
 
-float convert_planet_age(float planet, const long input)
+/* Number of seconds in a Earth year */
+#define EARTH 31557600
+
+/* Orbital period of each planet, in Earth years */
+#define MERCURY (0.2408467 * EARTH)
+#define VENUS (0.61519726 * EARTH)
+#define MARS (1.8808158 * EARTH)
+#define JUPITER (11.862615 * EARTH)
+#define SATURN (29.447498 * EARTH)
+#define URANUS (84.016846 * EARTH)
+#define NEPTUNE (164.79132 * EARTH)
+
+float convert_planet_age(Planets planet, const long input)
 {
-   return input / planet;
+   switch (planet) {
```

Comment:
What you do is use the enum value as a table index. Since the enumerated values start at 0 and increment by 1, they are natural array keys. So, something like this:

```c
static const float factors = {
    543 // mercury,
    123, // venus
    // ...
    56789 // neptune
};

// ...

float convert_planet_age(Planets planet, const long input) {
   return input / factors[planet];
}
```